lib/Core/Executor.cpp

```diff
@@ -2566,16 +2686,92 @@ void Executor::executeInstruction(ExecutionState &state, KInstruction *ki) {
     break;
   }
 #endif
+  case Instruction::InsertElement: {
+    InsertElementInst *iei = cast<InsertElementInst>(i);
+    ref<Expr> vec = eval(ki, 0, state).value;
+    ref<Expr> newElt = eval(ki, 1, state).value;
+    ref<Expr> idx = eval(ki, 2, state).value;
+
+    if (!isa<ConstantExpr>(idx)) {
```

Comment:
As you need a `ConstantExpr` anyway later, do a `dyn_cast<>()` and check for `nullptr`.